File: backend/src/main/java/com/relief/service/training/CertificationTrackingService.java
package com.relief.service.training;

import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Service;

import java.time.LocalDateTime;
import java.util.*;

/**
 * Certification tracking service for training and certification requirements
 */
@Service
@RequiredArgsConstructor
@Slf4j
public class CertificationTrackingService {

    public Certification createCertification(String name, String description, String category, 
                                           String issuingOrganization, int validityMonths, 
                                           List<String> requiredSkills, Map<String, Object> requirements) {
        Certification certification = new Certification();
        certification.setId(UUID.randomUUID().toString());
        certification.setName(name);
        certification.setDescription(description);
        certification.setCategory(category);
        certification.setIssuingOrganization(issuingOrganization);
        certification.setValidityMonths(validityMonths);
        certification.setRequiredSkills(requiredSkills);
        certification.setRequirements(requirements);
        certification.setCreatedAt(LocalDateTime.now());
        certification.setIsActive(true);
        
        log.info("Created certification: {}", certification.getId());
        return certification;
    }

    public UserCertification assignCertification(String userId, String certificationId, 
                                               LocalDateTime assignedDate, LocalDateTime expiryDate) {
        UserCertification userCert = new UserCertification();
        userCert.setId(UUID.randomUUID().toString());
        userCert.setUserId(userId);
        userCert.setCertificationId(certificationId);
        userCert.setAssignedDate(assignedDate);
        userCert.setExpiryDate(expiryDate);
        userCert.setStatus(CertificationStatus.ASSIGNED);
        userCert.setCreatedAt(LocalDateTime.now());
        
        log.info("Assigned certification {} to user {}", certificationId, userId);
        return userCert;
    }

    public UserCertification completeCertification(String userCertificationId, String completionMethod, 
                                                  String completionNotes, String verifiedBy) {
        UserCertification userCert = new UserCertification();
        userCert.setId(userCertificationId);
        userCert.setStatus(CertificationStatus.COMPLETED);
        userCert.setCompletionDate(LocalDateTime.now());
        userCert.setCompletionMethod(completionMethod);
        userCert.setCompletionNotes(completionNotes);
        userCert.setVerifiedBy(verifiedBy);
        userCert.setUpdatedAt(LocalDateTime.now());
        
        log.info("Completed certification: {}", userCertificationId);
        return userCert;
    }

    public UserCertification renewCertification(String userCertificationId, LocalDateTime newExpiryDate, 
                                              String renewalNotes) {
        UserCertification userCert = new UserCertification();
        userCert.setId(userCertificationId);
        userCert.setStatus(CertificationStatus.RENEWED);
        userCert.setExpiryDate(newExpiryDate);
        userCert.setRenewalDate(LocalDateTime.now());
        userCert.setRenewalNotes(renewalNotes);
        userCert.setUpdatedAt(LocalDateTime.now());
        
        log.info("Renewed certification: {}", userCertificationId);
        return userCert;
    }

    public UserCertification revokeCertification(String userCertificationId, String reason, String revokedBy) {
        UserCertification userCert = new UserCertification();
        userCert.setId(userCertificationId);
        userCert.setStatus(CertificationStatus.REVOKED);
        userCert.setRevocationDate(LocalDateTime.now());
        userCert.setRevocationReason(reason);
        userCert.setRevokedBy(revokedBy);
        userCert.setUpdatedAt(LocalDateTime.now());
        
        log.info("Revoked certification: {}", userCertificationId);
        return userCert;
    }

    public List<UserCertification> getUserCertifications(String userId, CertificationStatus status) {
        // Implementation for getting user certifications
        return Collections.emptyList();
    }

    public List<UserCertification> getExpiringCertifications(int daysBeforeExpiry) {
        // Implementation for getting expiring certifications
        return Collections.emptyList();
    }

    public List<UserCertification> getExpiredCertifications() {
        // Implementation for getting expired certifications
        return Collections.emptyList();
    }

    public List<Certification> getCertifications(String category, boolean isActive) {
        // Implementation for getting certifications
        return Collections.emptyList();
    }

    public Certification getCertification(String certificationId) {
        // Implementation for getting certification details
        return new Certification();
    }

    public UserCertification getUserCertification(String userCertificationId) {
        // Implementation for getting user certification details
        return new UserCertification();
    }

    public CertificationRequirement createRequirement(String certificationId, String requirementType, 
                                                    String description, boolean isMandatory, 
                                                    Map<String, Object> criteria) {
        CertificationRequirement requirement = new CertificationRequirement();
        requirement.setId(UUID.randomUUID().toString());
        requirement.setCertificationId(certificationId);
        requirement.setRequirementType(requirementType);
        requirement.setDescription(description);
        requirement.setIsMandatory(isMandatory);
        requirement.setCriteria(criteria);
        requirement.setCreatedAt(LocalDateTime.now());
        
        log.info("Created certification requirement: {}", requirement.getId());
        return requirement;
    }

    public List<CertificationRequirement> getCertificationRequirements(String certificationId) {
        // Implementation for getting certification requirements
        return Collections.emptyList();
    }

    public CertificationProgress trackProgress(String userCertificationId, String requirementId, 
                                             String status, String notes) {
        CertificationProgress progress = new CertificationProgress();
        progress.setId(UUID.randomUUID().toString());
        progress.setUserCertificationId(userCertificationId);
        progress.setRequirementId(requirementId);
        progress.setStatus(status);
        progress.setNotes(notes);
        progress.setUpdatedAt(LocalDateTime.now());
        
        log.info("Updated certification progress: {}", progress.getId());
        return progress;
    }

    public List<CertificationProgress> getCertificationProgress(String userCertificationId) {
        // Implementation for getting certification progress
        return Collections.emptyList();
    }

    public CertificationReport generateReport(String userId, String certificationId, 
                                            LocalDateTime startDate, LocalDateTime endDate) {
        CertificationReport report = new CertificationReport();
        report.setId(UUID.randomUUID().toString());
        report.setUserId(userId);
        report.setCertificationId(certificationId);
        report.setStartDate(startDate);
        report.setEndDate(endDate);
        report.setGeneratedAt(LocalDateTime.now());
        report.setTotalCertifications(0);
        report.setCompletedCertifications(0);
        report.setExpiredCertifications(0);
        report.setExpiringCertifications(0);
        report.setComplianceRate(0.0);
        
        log.info("Generated certification report: {}", report.getId());
        return report;
    }

    public List<CertificationAlert> getCertificationAlerts(String userId) {
        // Implementation for getting certification alerts
        return Collections.emptyList();
    }

    public CertificationAnalytics getCertificationAnalytics(String organizationId) {
        CertificationAnalytics analytics = new CertificationAnalytics();
        analytics.setOrganizationId(organizationId);
        analytics.setTotalCertifications(0);
        analytics.setActiveCertifications(0);
        analytics.setExpiredCertifications(0);
        analytics.setExpiringCertifications(0);
        analytics.setComplianceRate(0.0);
        analytics.setAverageCompletionTime(0);
        analytics.setPopularCertifications(Collections.emptyList());
        analytics.setCertificationTrends(Collections.emptyList());
        
        return analytics;
    }

    public void sendExpiryNotifications() {
        log.info("Sending certification expiry notifications");
    }

    public void sendRenewalReminders() {
        log.info("Sending certification renewal reminders");
    }

    // Data classes
    public static class Certification {
        private String id;
        private String name;
        private String description;
        private String category;
        private String issuingOrganization;
        private int validityMonths;
        private List<String> requiredSkills;
        private Map<String, Object> requirements;
        private LocalDateTime createdAt;
        private boolean isActive;

        // Getters and setters
        public String getId() { return id; }
        public void setId(String id) { this.id = id; }

        public String getName() { return name; }
        public void setName(String name) { this.name = name; }

        public String getDescription() { return description; }
        public void setDescription(String description) { this.description = description; }

        public String getCategory() { return category; }
        public void setCategory(String category) { this.category = category; }

        public String getIssuingOrganization() { return issuingOrganization; }
        public void setIssuingOrganization(String issuingOrganization) { this.issuingOrganization = issuingOrganization; }

        public int getValidityMonths() { return validityMonths; }
        public void setValidityMonths(int validityMonths) { this.validityMonths = validityMonths; }

        public List<String> getRequiredSkills() { return requiredSkills; }
        public void setRequiredSkills(List<String> requiredSkills) { this.requiredSkills = requiredSkills; }

        public Map<String, Object> getRequirements() { return requirements; }
        public void setRequirements(Map<String, Object> requirements) { this.requirements = requirements; }

        public LocalDateTime getCreatedAt() { return createdAt; }
        public void setCreatedAt(LocalDateTime createdAt) { this.createdAt = createdAt; }

        public boolean isActive() { return isActive; }
        public void setActive(boolean active) { isActive = active; }
        public void setIsActive(boolean active) { isActive = active; }
    }

    public static class UserCertification {
        private String id;
        private String userId;
        private String certificationId;
        private LocalDateTime assignedDate;
        private LocalDateTime completionDate;
        private LocalDateTime expiryDate;
        private LocalDateTime renewalDate;
        private LocalDateTime revocationDate;
        private CertificationStatus status;
        private String completionMethod;
        private String completionNotes;
        private String verifiedBy;
        private String renewalNotes;
        private String revocationReason;
        private String revokedBy;
        private LocalDateTime createdAt;
        private LocalDateTime updatedAt;

        // Getters and setters
        public String getId() { return id; }
        public void setId(String id) { this.id = id; }

        public String getUserId() { return userId; }
        public void setUserId(String userId) { this.userId = userId; }

        public String getCertificationId() { return certificationId; }
        public void setCertificationId(String certificationId) { this.certificationId = certificationId; }

        public LocalDateTime getAssignedDate() { return assignedDate; }
        public void setAssignedDate(LocalDateTime assignedDate) { this.assignedDate = assignedDate; }

        public LocalDateTime getCompletionDate() { return completionDate; }
        public void setCompletionDate(LocalDateTime completionDate) { this.completionDate = completionDate; }

        public LocalDateTime getExpiryDate() { return expiryDate; }
        public void setExpiryDate(LocalDateTime expiryDate) { this.expiryDate = expiryDate; }

        public LocalDateTime getRenewalDate() { return renewalDate; }
        public void setRenewalDate(LocalDateTime renewalDate) { this.renewalDate = renewalDate; }

        public LocalDateTime getRevocationDate() { return revocationDate; }
        public void setRevocationDate(LocalDateTime revocationDate) { this.revocationDate = revocationDate; }

        public CertificationStatus getStatus() { return status; }
        public void setStatus(CertificationStatus status) { this.status = status; }

        public String getCompletionMethod() { return completionMethod; }
        public void setCompletionMethod(String completionMethod) { this.completionMethod = completionMethod; }

        public String getCompletionNotes() { return completionNotes; }
        public void setCompletionNotes(String completionNotes) { this.completionNotes = completionNotes; }

        public String getVerifiedBy() { return verifiedBy; }
        public void setVerifiedBy(String verifiedBy) { this.verifiedBy = verifiedBy; }

        public String getRenewalNotes() { return renewalNotes; }
        public void setRenewalNotes(String renewalNotes) { this.renewalNotes = renewalNotes; }

        public String getRevocationReason() { return revocationReason; }
        public void setRevocationReason(String revocationReason) { this.revocationReason = revocationReason; }

        public String getRevokedBy() { return revokedBy; }
        public void setRevokedBy(String revokedBy) { this.revokedBy = revokedBy; }

        public LocalDateTime getCreatedAt() { return createdAt; }
        public void setCreatedAt(LocalDateTime createdAt) { this.createdAt = createdAt; }

        public LocalDateTime getUpdatedAt() { return updatedAt; }
        public void setUpdatedAt(LocalDateTime updatedAt) { this.updatedAt = updatedAt; }
    }

    public static class CertificationRequirement {
        private String id;
        private String certificationId;
        private String requirementType;
        private String description;
        private boolean isMandatory;
        private Map<String, Object> criteria;
        private LocalDateTime createdAt;

        // Getters and setters
        public String getId() { return id; }
        public void setId(String id) { this.id = id; }

        public String getCertificationId() { return certificationId; }
        public void setCertificationId(String certificationId) { this.certificationId = certificationId; }

        public String getRequirementType() { return requirementType; }
        public void setRequirementType(String requirementType) { this.requirementType = requirementType; }

        public String getDescription() { return description; }
        public void setDescription(String description) { this.description = description; }

        public boolean isMandatory() { return isMandatory; }
        public void setMandatory(boolean mandatory) { isMandatory = mandatory; }
        public void setIsMandatory(boolean mandatory) { this.setMandatory(mandatory); }

        public Map<String, Object> getCriteria() { return criteria; }
        public void setCriteria(Map<String, Object> criteria) { this.criteria = criteria; }

        public LocalDateTime getCreatedAt() { return createdAt; }
        public void setCreatedAt(LocalDateTime createdAt) { this.createdAt = createdAt; }
    }

    public static class CertificationProgress {
        private String id;
        private String userCertificationId;
        private String requirementId;
        private String status;
        private String notes;
        private LocalDateTime updatedAt;

        // Getters and setters
        public String getId() { return id; }
        public void setId(String id) { this.id = id; }

        public String getUserCertificationId() { return userCertificationId; }
        public void setUserCertificationId(String userCertificationId) { this.userCertificationId = userCertificationId; }

        public String getRequirementId() { return requirementId; }
        public void setRequirementId(String requirementId) { this.requirementId = requirementId; }

        public String getStatus() { return status; }
        public void setStatus(String status) { this.status = status; }

        public String getNotes() { return notes; }
        public void setNotes(String notes) { this.notes = notes; }

        public LocalDateTime getUpdatedAt() { return updatedAt; }
        public void setUpdatedAt(LocalDateTime updatedAt) { this.updatedAt = updatedAt; }
    }

    public static class CertificationReport {
        private String id;
        private String userId;
        private String certificationId;
        private LocalDateTime startDate;
        private LocalDateTime endDate;
        private LocalDateTime generatedAt;
        private int totalCertifications;
        private int completedCertifications;
        private int expiredCertifications;
        private int expiringCertifications;
        private double complianceRate;

        // Getters and setters
        public String getId() { return id; }
        public void setId(String id) { this.id = id; }

        public String getUserId() { return userId; }
        public void setUserId(String userId) { this.userId = userId; }

        public String getCertificationId() { return certificationId; }
        public void setCertificationId(String certificationId) { this.certificationId = certificationId; }

        public LocalDateTime getStartDate() { return startDate; }
        public void setStartDate(LocalDateTime startDate) { this.startDate = startDate; }

        public LocalDateTime getEndDate() { return endDate; }
        public void setEndDate(LocalDateTime endDate) { this.endDate = endDate; }

        public LocalDateTime getGeneratedAt() { return generatedAt; }
        public void setGeneratedAt(LocalDateTime generatedAt) { this.generatedAt = generatedAt; }

        public int getTotalCertifications() { return totalCertifications; }
        public void setTotalCertifications(int totalCertifications) { this.totalCertifications = totalCertifications; }

        public int getCompletedCertifications() { return completedCertifications; }
        public void setCompletedCertifications(int completedCertifications) { this.completedCertifications = completedCertifications; }

        public int getExpiredCertifications() { return expiredCertifications; }
        public void setExpiredCertifications(int expiredCertifications) { this.expiredCertifications = expiredCertifications; }

        public int getExpiringCertifications() { return expiringCertifications; }
        public void setExpiringCertifications(int expiringCertifications) { this.expiringCertifications = expiringCertifications; }

        public double getComplianceRate() { return complianceRate; }
        public void setComplianceRate(double complianceRate) { this.complianceRate = complianceRate; }
    }

    public static class CertificationAlert {
        private String id;
        private String userId;
        private String certificationId;
        private String alertType;
        private String message;
        private LocalDateTime createdAt;
        private boolean isRead;

        // Getters and setters
        public String getId() { return id; }
        public void setId(String id) { this.id = id; }

        public String getUserId() { return userId; }
        public void setUserId(String userId) { this.userId = userId; }

        public String getCertificationId() { return certificationId; }
        public void setCertificationId(String certificationId) { this.certificationId = certificationId; }

        public String getAlertType() { return alertType; }
        public void setAlertType(String alertType) { this.alertType = alertType; }

        public String getMessage() { return message; }
        public void setMessage(String message) { this.message = message; }

        public LocalDateTime getCreatedAt() { return createdAt; }
        public void setCreatedAt(LocalDateTime createdAt) { this.createdAt = createdAt; }

        public boolean isRead() { return isRead; }
        public void setRead(boolean read) { isRead = read; }
    }

    public static class CertificationAnalytics {
        private String organizationId;
        private int totalCertifications;
        private int activeCertifications;
        private int expiredCertifications;
        private int expiringCertifications;
        private double complianceRate;
        private int averageCompletionTime;
        private List<String> popularCertifications;
        private List<Map<String, Object>> certificationTrends;

        // Getters and setters
        public String getOrganizationId() { return organizationId; }
        public void setOrganizationId(String organizationId) { this.organizationId = organizationId; }

        public int getTotalCertifications() { return totalCertifications; }
        public void setTotalCertifications(int totalCertifications) { this.totalCertifications = totalCertifications; }

        public int getActiveCertifications() { return activeCertifications; }
        public void setActiveCertifications(int activeCertifications) { this.activeCertifications = activeCertifications; }

        public int getExpiredCertifications() { return expiredCertifications; }
        public void setExpiredCertifications(int expiredCertifications) { this.expiredCertifications = expiredCertifications; }

        public int getExpiringCertifications() { return expiringCertifications; }
        public void setExpiringCertifications(int expiringCertifications) { this.expiringCertifications = expiringCertifications; }

        public double getComplianceRate() { return complianceRate; }
        public void setComplianceRate(double complianceRate) { this.complianceRate = complianceRate; }

        public int getAverageCompletionTime() { return averageCompletionTime; }
        public void setAverageCompletionTime(int averageCompletionTime) { this.averageCompletionTime = averageCompletionTime; }

        public List<String> getPopularCertifications() { return popularCertifications; }
        public void setPopularCertifications(List<String> popularCertifications) { this.popularCertifications = popularCertifications; }

        public List<Map<String, Object>> getCertificationTrends() { return certificationTrends; }
        public void setCertificationTrends(List<Map<String, Object>> certificationTrends) { this.certificationTrends = certificationTrends; }
    }

    public enum CertificationStatus {
        ASSIGNED, IN_PROGRESS, COMPLETED, EXPIRED, RENEWED, REVOKED
    }
}

File: frontend/src/services/certificationTrackingService.ts
import { apiService } from './api';

export interface Certification {
  id: string;
  name: string;
  description: string;
  category: string;
  issuingOrganization: string;
  validityMonths: number;
  requiredSkills: string[];
  requirements: Record<string, any>;
  createdAt: string;
  isActive: boolean;
}

export interface UserCertification {
  id: string;
  userId: string;
  certificationId: string;
  assignedDate: string;
  completionDate?: string;
  expiryDate: string;
  renewalDate?: string;
  revocationDate?: string;
  status: 'ASSIGNED' | 'IN_PROGRESS' | 'COMPLETED' | 'EXPIRED' | 'RENEWED' | 'REVOKED';
  completionMethod?: string;
  completionNotes?: string;
  verifiedBy?: string;
  renewalNotes?: string;
  revocationReason?: string;
  revokedBy?: string;
  createdAt: string;
  updatedAt: string;
}

export interface CertificationRequirement {
  id: string;
  certificationId: string;
  requirementType: string;
  description: string;
  isMandatory: boolean;
  criteria: Record<string, any>;
  createdAt: string;
}

export interface CertificationProgress {
  id: string;
  userCertificationId: string;
  requirementId: string;
  status: string;
  notes?: string;
  updatedAt: string;
}

export interface CertificationReport {
  id: string;
  userId: string;
  certificationId: string;
  startDate: string;
  endDate: string;
  generatedAt: string;
  totalCertifications: number;
  completedCertifications: number;
  expiredCertifications: number;
  expiringCertifications: number;
  complianceRate: number;
}

export interface CertificationAlert {
  id: string;
  userId: string;
  certificationId: string;
  alertType: string;
  message: string;
  createdAt: string;
  isRead: boolean;
}

export interface CertificationAnalytics {
  organizationId: string;
  totalCertifications: number;
  activeCertifications: number;
  expiredCertifications: number;
  expiringCertifications: number;
  complianceRate: number;
  averageCompletionTime: number;
  popularCertifications: string[];
  certificationTrends: Array<Record<string, any>>;
}

export interface CreateCertificationRequest {
  name: string;
  description: string;
  category: string;
  issuingOrganization: string;
  validityMonths: number;
  requiredSkills: string[];
  requirements: Record<string, any>;
}

export interface AssignCertificationRequest {
  userId: string;
  certificationId: string;
  assignedDate: string;
  expiryDate: string;
}

export interface CompleteCertificationRequest {
  completionMethod: string;
  completionNotes?: string;
}

export interface RenewCertificationRequest {
  newExpiryDate: string;
  renewalNotes?: string;
}

export interface RevokeCertificationRequest {
  reason: string;
}

export interface CreateRequirementRequest {
  certificationId: string;
  requirementType: string;
  description: string;
  isMandatory: boolean;
  criteria: Record<string, any>;
}

export interface TrackProgressRequest {
  userCertificationId: string;
  requirementId: string;
  status: string;
  notes?: string;
}

export interface GenerateReportRequest {
  userId: string;
  certificationId?: string;
  startDate: string;
  endDate: string;
}

class CertificationTrackingService {
  private baseUrl = '/certification-tracking';

  async createCertification(request: CreateCertificationRequest): Promise<Certification> {
    return apiService.post(`${this.baseUrl}/certifications`, request);
  }

  async assignCertification(request: AssignCertificationRequest): Promise<UserCertification> {
    return apiService.post(`${this.baseUrl}/user-certifications`, request);
  }

  async completeCertification(userCertificationId: string, request: CompleteCertificationRequest): Promise<UserCertification> {
    return apiService.post(`${this.baseUrl}/user-certifications/${userCertificationId}/complete`, request);
  }

  async renewCertification(userCertificationId: string, request: RenewCertificationRequest): Promise<UserCertification> {
    return apiService.post(`${this.baseUrl}/user-certifications/${userCertificationId}/renew`, request);
  }

  async revokeCertification(userCertificationId: string, request: RevokeCertificationRequest): Promise<UserCertification> {
    return apiService.post(`${this.baseUrl}/user-certifications/${userCertificationId}/revoke`, request);
  }

  async getUserCertifications(userId: string, status?: string): Promise<UserCertification[]> {
    return apiService.get(`${this.baseUrl}/user-certifications`, { userId, status });
  }

  async getExpiringCertifications(daysBeforeExpiry: number = 30): Promise<UserCertification[]> {
    return apiService.get(`${this.baseUrl}/certifications/expiring`, { daysBeforeExpiry });
  }

  async getExpiredCertifications(): Promise<UserCertification[]> {
    return apiService.get(`${this.baseUrl}/certifications/expired`);
  }

  async getCertifications(filters: {
    category?: string;
    isActive?: boolean;
  } = {}): Promise<Certification[]> {
    return apiService.get(`${this.baseUrl}/certifications`, filters);
  }

  async getCertification(certificationId: string): Promise<Certification> {
    return apiService.get(`${this.baseUrl}/certifications/${certificationId}`);
  }

  async getUserCertification(userCertificationId: string): Promise<UserCertification> {
    return apiService.get(`${this.baseUrl}/user-certifications/${userCertificationId}`);
  }

  async createRequirement(request: CreateRequirementRequest): Promise<CertificationRequirement> {
    return apiService.post(`${this.baseUrl}/requirements`, request);
  }

  async getCertificationRequirements(certificationId: string): Promise<CertificationRequirement[]> {
    return apiService.get(`${this.baseUrl}/certifications/${certificationId}/requirements`);
  }

  async trackProgress(request: TrackProgressRequest): Promise<CertificationProgress> {
    return apiService.post(`${this.baseUrl}/progress`, request);
  }

  async getCertificationProgress(userCertificationId: string): Promise<CertificationProgress[]> {
    return apiService.get(`${this.baseUrl}/user-certifications/${userCertificationId}/progress`);
  }

  async generateReport(request: GenerateReportRequest): Promise<CertificationReport> {
    return apiService.post(`${this.baseUrl}/reports`, request);
  }

  async getCertificationAlerts(userId: string): Promise<CertificationAlert[]> {
    return apiService.get(`${this.baseUrl}/alerts`, { userId });
  }

  async getCertificationAnalytics(organizationId: string): Promise<CertificationAnalytics> {
    return apiService.get(`${this.baseUrl}/analytics`, { organizationId });
  }

  async sendExpiryNotifications(): Promise<void> {
    return apiService.post(`${this.baseUrl}/notifications/expiry`);
  }

  async sendRenewalReminders(): Promise<void> {
    return apiService.post(`${this.baseUrl}/notifications/renewal`);
  }
}

export const certificationTrackingService = new CertificationTrackingService();


